lib/shopify_api/resources/order.rb

```diff
@@ -7,7 +7,7 @@ def close;  load_attributes_from_response(post(:close, {}, only_id)); end
     def open;   load_attributes_from_response(post(:open, {}, only_id)); end
 
     def cancel(options = {})
-      load_attributes_from_response(post(:cancel, options, only_id))
+      load_attributes_from_response(post(:cancel, {}, options.to_json))
```

Comment:
[`only_id`](https://github.com/Shopify/shopify_api/blob/e729054363fd30e584a5fb2fb6b2e92cd41d6e73/lib/shopify_api/resources/base.rb#L81-L83) is a function that would encode only the records ID and make that the contents of the request body. (I only saw empty JSON in the request body)

The ID is encoded in the path so we don't need to worry about it, and our requests don't need to send that; only the data that is going to be acted upon.